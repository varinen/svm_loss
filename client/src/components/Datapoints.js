import React, {Component} from 'react';
import MathJax from 'react-mathjax';
import {connect} from "react-redux";
import {getData} from "../actions/data";

const formulas = {
    f1: `x_i`,
    f2: `y_i`,
    f3: `f(x_i;W,b)=Wx_i+b`,
    f4: `s[0] = x[0] * W[0,0] + x[1] * W[0,1] + b[0]`,
    f5: `L_i`
};

export class Datapoints extends Component {
    getData = () => {
        this.props.getData(1)
    };

    componentDidMount() {
        this.props.getData();
    }

    render() {
        return (
            <MathJax.Provider>
                <div className="datapoints card shadow">
                    <div className="card-header py-3">
                        <h6 className="m-0 font-weight-bold text-primary">
                            Data Point And Loss Values
                        </h6>
                    </div>
                    <div className="card-body">
                        <article>
                            <p>
                                Each row is loss due to one datapoint. The
                                first
                                three
                                columns are the 2D data <MathJax.Node inline
                                                                      formula={formulas.f1}/> and
                                the label <MathJax.Node inline
                                                        formula={formulas.f2}/>.

                                The next three columns are the three class
                                scores
                                from
                                each classifier <MathJax.Node inline
                                                              formula={formulas.f3}/> (E.g., <MathJax.Node
                                inline formula={formulas.f4}/>). The last
                                column is
                                the
                                data loss for a single example, <MathJax.Node
                                inline
                                formula={formulas.f5}/>

                            </p>


                            <div>

                            </div>
                        </article>
                    </div>
                </div>
            </MathJax.Provider>

        )
    }

}

export default connect(state=> (state.data), {getData})(Datapoints);
