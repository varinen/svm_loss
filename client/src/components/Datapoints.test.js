import React from 'react';
import {shallow} from 'enzyme';
import {Datapoints} from './Datapoints';

describe('Datapoints', () => {
    const datapoints = shallow(<Datapoints/>);

    it('renders correctly', () => {
        expect(datapoints).toMatchSnapshot();
    });
});

