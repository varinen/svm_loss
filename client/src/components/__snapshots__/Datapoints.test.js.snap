// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datapoints renders correctly 1`] = `
<MathJaxProvider
  options={
    Object {
      "showMathMenu": false,
      "showMathMenuMSIE": false,
      "tex2jax": Object {
        "inlineMath": Array [],
      },
    }
  }
  script="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-MML-AM_CHTML"
>
  <div
    className="datapoints card shadow"
  >
    <div
      className="card-header py-3"
    >
      <h6
        className="m-0 font-weight-bold text-primary"
      >
        Data Point And Loss Values
      </h6>
    </div>
    <div
      className="card-body"
    >
      <article>
        <p>
          Each row is loss due to one datapoint. The first three columns are the 2D data 
          <MathJaxNode
            formula="x_i"
            inline={true}
          />
           and the label 
          <MathJaxNode
            formula="y_i"
            inline={true}
          />
          . The next three columns are the three class scores from each classifier 
          <MathJaxNode
            formula="f(x_i;W,b)=Wx_i+b"
            inline={true}
          />
           (E.g., 
          <MathJaxNode
            formula="s[0] = x[0] * W[0,0] + x[1] * W[0,1] + b[0]"
            inline={true}
          />
          ). The last column is the data loss for a single example, 
          <MathJaxNode
            formula="L_i"
            inline={true}
          />
        </p>
        <div />
      </article>
    </div>
  </div>
</MathJaxProvider>
`;
